text/0000-euclidean-modulo.md

```diff
@@ -0,0 +1,94 @@
+- Feature Name: euclidean_modulo
+- Start Date: 2017-10-09
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+This RFC proposes the addition of a modulo method with more useful and mathematically regular properties over the built-in remainder `%` operator when the dividend or divisor is negative, along with the associated division method.
+
+For previous discussion, see: https://internals.rust-lang.org/t/mathematical-modulo-operator/5952.
+
+# Motivation
+[motivation]: #motivation
+
+The behaviour of division and modulo, as implemented by Rust's (truncated) division `/` and remainder (or truncated modulo) `%` operators, with respect to negative operands is unintuitive and has fewer useful mathematical properties than that of other varieties of division and modulo, such as flooring and Euclidean[[1]](https://dl.acm.org/citation.cfm?doid=128861.128862). While there are good reasons for this design decision[[2]](https://mail.mozilla.org/pipermail/rust-dev/2013-April/003786.html), having convenient access to a modulo operation, in addition to the remainder is very useful, and has often been requested[[3]](https://mail.mozilla.org/pipermail/rust-dev/2013-April/003680.html)[[4]](https://github.com/rust-lang/rust/issues/13909)[[5]](https://stackoverflow.com/questions/31210357/is-there-a-modulus-not-remainder-function-operation)[[6]](https://users.rust-lang.org/t/proper-modulo-support/903)[[7]](https://www.reddit.com/r/rust/comments/3yoo1q/remainder/).
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+```rust
+// Comparison of the behaviour of Rust's truncating division
+// and remainder, vs Euclidean division & modulo.
+(-8 / 3,      -8 % 3)       // (-2, -2)
+(-8.div_e(3), -8.mod_e(3))  // (-3,  1)
```

Comment:
`.` has higher precedence than unary minus. This will be evaluated as `-( 8.mod_e(3) )` i.e. `-2`. You want `(-8).mod_e(3)`.

Reply:
Good catch, thanks!